lib/util/sfBrowserBase.class.php

```diff
@@ -778,8 +778,10 @@ public function doClickElement(DOMElement $item, $arguments = array(), $options
     }
     else if ('button' == $item->nodeName || ('input' == $item->nodeName && in_array($item->getAttribute('type'), array('submit', 'button', 'image'))))
     {
-      // add the item's value to the arguments
-      $this->parseArgumentAsArray($item->getAttribute('name'), $item->getAttribute('value'), $arguments);
+      // add the item's value to the arguments if name is provided
+      if ($item->getAttribute('name')) {
```

Comment:
Updated. I'm sorry for making a mess here, I didn't know my traits support commit would go straight into this pull request.